src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jpa/entity/Referral.kt

```diff
@@ -79,7 +79,7 @@ class Referral(
     joinColumns = [JoinColumn(name = "referral_id")],
     inverseJoinColumns = [JoinColumn(name = "service_category_id")]
   )
-  var selectedServiceCategories: Set<ServiceCategory>? = null,
+  var selectedServiceCategories: MutableSet<ServiceCategory>? = null,
```

Comment:
As a mutable list can that be set to non nullable and initialised with an empty set?

Reply:
I believe we still want to maintain the concept of a null. This indicates that the user has not selected anything yet, which is different than selecting nothing.

Reply:
for collection types spring returns empty collections for lookups anyway, so we actually have some code in the DTO layer to translate empty collections into null values where they don't make sense to be empty collections. it's a bit of a mess tbh. but either way it doesn't really matter here. let's make the changes as minimal as possible and leave the init to null.